src/lib/downloadReachability.js

```diff
@@ -0,0 +1,93 @@
+const DEFAULT_DOWNLOAD_HOST_PREFIX = 'https://dl.';
+const DOWNLOAD_NOTIFICATION_SHOWN_KEY = 'download_host_notification_shown';
+const DOWNLOAD_HOST_FALLBACK_KEY = 'download_host_fallback';
+const DEFAULT_HOST_REGEX = /^https:\/\/dl\d+\./;
+
+/**
+ * Checks if the url is a download host, but not the default download host.
+ *
+ * @public
+ * @param {string} downloadUrl - Content download URL, may either be a template or an actual URL
+ * @return {boolean} - HTTP response
+ */
+export function isCustomDownloadHost(downloadUrl) {
+    return (
+        downloadUrl && !downloadUrl.startsWith(DEFAULT_DOWNLOAD_HOST_PREFIX) && !!downloadUrl.match(DEFAULT_HOST_REGEX)
+    );
+}
+
+/**
+ * Replaces the hostname of a download URL with the default hostname, https://dl.
+ *
+ * @private
+ * @param {string} downloadUrl - Content download URL, may either be a template or an actual URL
+ * @return {string} - The updated download URL
+ */
+export function replaceDownloadHostWithDefault(downloadUrl) {
+    return downloadUrl.replace(DEFAULT_HOST_REGEX, DEFAULT_DOWNLOAD_HOST_PREFIX);
```

Comment:
Some AWS or accelerated hosts have weird patterns like `https://ec2apne1.boxcloud.com`, I'm not sure your regex would capture that

Reply:
womp, that is good information to know. Do you think its safe to check `startsWith(dl#.) || endsWith('boxcloud.com')`? Excuse the hand-wavey pseudocode.